common/src/View/InputEvent.cpp

```diff
@@ -142,21 +143,18 @@ namespace TrenchBroom {
                         m_queue.enqueueEvent(std::make_unique<CancelEvent>());
                         m_dragging = false;
 
-                        // Synthesize a click event if the drag distance was not exceeded.
-                        if (!isDrag(posX, posY)) {
-                            m_queue.enqueueEvent(std::make_unique<MouseEvent>(MouseEvent::Type::Click, button, wheelAxis, m_lastClickX, m_lastClickY, scrollDistance));
-                        }
+                        // Synthesize a click event
+                        m_queue.enqueueEvent(std::make_unique<MouseEvent>(MouseEvent::Type::Click, button, wheelAxis, m_lastClickX, m_lastClickY, scrollDistance));
                     } else {
                         m_queue.enqueueEvent(std::make_unique<MouseEvent>(MouseEvent::Type::DragEnd, button, wheelAxis, posX, posY, scrollDistance));
                         m_dragging = false;
                     }
-                } else {
-                    // Synthesize a click event if the drag distance was not exceeded.
-                    if (!isDrag(posX, posY)) {
-                        m_queue.enqueueEvent(std::make_unique<MouseEvent>(MouseEvent::Type::Click, button, wheelAxis, m_lastClickX, m_lastClickY, scrollDistance));
-                    }
+                } else if (!m_nextMouseUpIsDblClick) {
+                    // Synthesize a click event
```

Comment:
ditto, why is the isDrag check removed?

Reply:
Here it really didn't make sense: `m_dragging` is false, so the drag distance hasn't been exceeded for sure - `isDrag` would always return false here.